tests/tests/test_functions.py

```diff
@@ -0,0 +1,19 @@
+import psycopg2
+
+from fixtures.glaredb import glaredb_connection, debug_path
+
+
+def test_scalar_parsing(
+    glaredb_connection: psycopg2.extensions.connection,
+):
+    for operation in ["create table t (x int);", "insert into t values (1), (2), (3);"]:
+        with glaredb_connection.cursor() as cur:
+            cur.execute(operation)
+
+    with glaredb_connection.cursor() as cur:
+        cur.execute("select siphash(x) as actual, siphash(arrow_cast(1, 'Int32')) as one from t")
+        result = cur.fetchall()
+
+    assert result[0][0] == result[0][1]
+    assert result[1][0] != result[1][1]
+    assert result[2][0] != result[2][1]
```

Comment:
not blocking, but I'd prefer if these were in the slts. They run on all protocols, while it appears that these are specific to using the postgres protocol.

Reply:
Fair enough, though I'm not particularly worried about execution level operations like this being impacted by service architecture aspects, and I _am_ more worried about the ways that multi-statement operations' statefullnes in SLTs can leak.